tests/python_helics/test_message_filter.py

```diff
@@ -23,42 +23,37 @@ def AddBroker(core_type="zmq", number_of_federates=1):
 def AddFederate(broker, core_type="zmq", count=1, deltat=1.0, name_prefix="fed"):
 
     # Create Federate Info object that describes the federate properties #
-    fedinfo = h.helicsFederateInfoCreate()
+    fedinfo = h.helicsCreateFederateInfo()
 
     # Set Federate name #
-    status = h.helicsFederateInfoSetFederateName(fedinfo, name_prefix)
-    assert status == 0
+    h.helicsFederateInfoSetCoreName(fedinfo, name_prefix)
 
     # Set core type from string #
-    status = h.helicsFederateInfoSetCoreTypeFromString(fedinfo, "zmq")
-    assert status == 0
+    h.helicsFederateInfoSetCoreTypeFromString(fedinfo, "zmq")
 
     # Federate init string #
     fedinitstring = "--broker=mainbroker --federates={}".format(count)
-    status = h.helicsFederateInfoSetCoreInitString(fedinfo, fedinitstring)
-    assert status == 0
+    h.helicsFederateInfoSetCoreInitString(fedinfo, fedinitstring)
 
     # Set the message interval (timedelta) for federate. Note th#
     # HELICS minimum message time interval is 1 ns and by default
     # it uses a time delta of 1 second. What is provided to the
     # setTimedelta routine is a multiplier for the default timedelta.
 
     # Set one second message interval #
-    status = h.helicsFederateInfoSetTimeDelta(fedinfo, deltat)
-    assert status == 0
+    h.helicsFederateInfoSetTimeProperty(fedinfo, h.helics_time_property_time_delta, deltat)
 
-    status = h.helicsFederateInfoSetLoggingLevel(fedinfo, 1)
-    assert status == 0
+    h.helicsFederateInfoSetIntegerProperty(fedinfo, h.helics_int_property_log_level, 1)
 
-    mFed = h.helicsCreateMessageFederate(fedinfo)
+
+    mFed = h.helicsCreateMessageFederate(name_prefix, fedinfo)
```

Comment:
You may want to free the federateInfo if you are done with it, otherwise it will be a small memory leak.

Reply:
I'm assuming it has to be freed only at the end of the program, and not immediately after creating the Federate. Is that correct?

Reply:
I don't believe the fedinfo maintains a pointer internal to the library like federates/cores/brokers so to have no leaks it would need to be freed at some point after you are done using it.    You can use it multiple times if you like.    it isn't that big of object and it is not likely to be created a whole bunch of times so it is a small issue if you forget but the test code should be good usage.  
So as long as you keep the object around you can free it whenever.

Reply:
Okay, I added free fedinfo functions along with the other free functions.